### PR TITLE
Submenu

### PR DESCRIPTION
**Pull request by angieelfin**

### CODE REVIEW

File: source/_header.html.slim [hunk 1]
```diff
@@ -0,0 +1,67 @@
+
```
Comment (angieelfin) [inline source/_header.html.slim]: empty line

File: source/_header.html.slim [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+.header.container-fluid
+  .head-nav
```
Comment (angieelfin) [inline source/_header.html.slim]: div class="head-nav"

File: source/_header.html.slim [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    .address
+      p WARSZAWA CENTRUM
```
Comment (angieelfin) [inline source/_header.html.slim]: p
  | Warszawa ...

File: source/_header.html.slim [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            a href='#' cennik 
+          li 
+            a href='#' galeria
+          li
+            a href='#' kontakt
+          li
+            a#diff-style href='#' lokalizacje
+            img.arrow src="images/arrow-color.png"
+        
+
```
Comment (angieelfin) [inline source/_header.html.slim]: Empty lines

File: source/stylesheets/all_styles.scss [hunk 5]
```diff
@@ -0,0 +1,21 @@
+// variables
+$grey: #484848;
+$light_grey: #ECECEC;
+$turquoise: #249799;
+$dark_blue: #195D73;
+
+@import "header";
+
+@font-face {
```
Comment (angieelfin) [inline source/stylesheets/all_styles.scss]: move to separate file

File: source/stylesheets/all_styles.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
+// variables
+$grey: #484848;
+$light_grey: #ECECEC;
+$turquoise: #249799;
+$dark_blue: #195D73;
+
+@import "header";
```
Comment (angieelfin) [inline source/stylesheets/all_styles.scss]: /layout/_header.scss

File: source/stylesheets/all_styles.scss [hunk 7]
```diff
@@ -0,0 +1,21 @@
+// variables
+$grey: #484848;
```
Comment (angieelfin) [inline source/stylesheets/all_styles.scss]: @import "variables"

File: source/stylesheets/header.scss [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+  margin: 5px 0;
+  p {
+    color: $grey;
+    font-weight: 400;
+    margin: 5px;
+  }
+  p.turquoise {
+    color: $turquoise;
+  }
+  .menu {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: .nav-menu

File: source/stylesheets/header.scss [hunk 9]
```diff
@@ -0,0 +1,162 @@
+
+//header navigation styling
+.head-nav {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: nav-header

File: source/stylesheets/header.scss [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+    margin: 5px;
+  }
+  p.turquoise {
+    color: $turquoise;
+  }
+  .menu {
+    display: inline-flex;
+    height: 0;
+    a {
+      font-size: 12px;
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: font-size into variable

File: source/stylesheets/header.scss [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+      font-weight: 400;
+      text-transform: uppercase;
+      color: $grey;
+      margin: 5px;
+      &:hover {
+        text-decoration: none;
+      }
+    }
+    a.turquoise {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: nav-register

File: source/stylesheets/header.scss [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+      color: $turquoise;
+    }
+    .lang {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: @import 'icons' => icon-language

File: source/stylesheets/header.scss [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+      height: 15px;
+      margin-bottom: 2px;
+    }
+    .divider {
+      height: 13px;
+      width: 13px;
+      margin-top: 5px;
+    }
+    .search {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: as above icon-search

File: source/stylesheets/header.scss [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+    }
+    a.turquoise {
+      color: $turquoise;
+    }
+    .lang {
+      height: 15px;
+      margin-bottom: 2px;
+    }
+    .divider {
+      height: 13px;
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: :before { content: '/'}

File: source/stylesheets/header.scss [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+      width: 13px;
+      margin-top: 5px;
+    }
+    .search {
+      height: 18px;
+      width: 18px;
+      margin-bottom: 5px;
+    }
+  }
+  .input-group {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: seperate class different as bootstrap

File: source/stylesheets/header.scss [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+    width: 140px;
+    height: 22px;
+    margin-right: 10px;
+    input, button {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: nav-search

File: source/stylesheets/header.scss [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+      box-shadow: none;
+      margin: 0;
+      padding: 8px;
+      height: 22px;
+      border-radius: 7px;
+    }
+    button {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: @import 'components/button' search-button

File: source/stylesheets/header.scss [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+    font-size: 12px;
+    &:focus {
+      border-color: $turquoise;
+      outline-color: none;
+    }
+  }
+}
+
+//logo
+.brand {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: nav-logo brand

File: source/stylesheets/header.scss [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+.navbar {
+  margin-top: 50px;
+  text-align: center;
+  .main-nav {
+    padding: 0;
+    list-style-type: none;
+    display: flex;
+    justify-content: space-between;
+    text-transform: uppercase;
+    font-size: 15px;
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: into variable

File: source/stylesheets/header.scss [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+  width: 270px;
+  height: 100%;
+  }
+}
+
+// main navigation styling
+.navbar {
+  margin-top: 50px;
+  text-align: center;
+  .main-nav {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: nav-main

File: source/stylesheets/header.scss [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+    cursor: pointer;
+    &:hover {
+      border-bottom: 3px solid $turquoise;
+    }
+    a {
+      text-decoration: none;
+      padding-bottom: 4px;
+      color: $grey;
+    }
+    #diff-style {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: remove id

Comment (angieelfin) [inline source/stylesheets/header.scss]: menu-item { .location} menu-item location li { ... last-child}

File: source/stylesheets/header.scss [hunk 22]
```diff
@@ -0,0 +1,162 @@
...
+      color: $turquoise;
+    }
+  }
+}
+.arrow, .arrow-color {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: remove arrow color

File: source/stylesheets/header.scss [hunk 23]
```diff
@@ -0,0 +1,162 @@
...
+   width: 12px;
+   margin-left: 3px;
+ }
+ // sub-navigation
+.nav-item {
+  position: relative;
+ }
+.nav-content {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: nav-menu nav-dropdown

File: source/stylesheets/header.scss [hunk 24]
```diff
@@ -0,0 +1,162 @@
...
+    &:hover {
+      border: none;
+    }
+  } 
+}
+// .nav-item:focus ~ .nav-content {
+//   display: block;
+// }
+
+@media screen and (max-width: 991px) {
```
Comment (angieelfin) [inline source/stylesheets/header.scss]: @import mediaqueries/tablet

File: source/_header.html.slim [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+          | Braci Wagów 20
+        p class="phone-number"
+          | tel. 223 009 400
+    div class="register-nav"
+      div class="input-group search-group"
+        input class="form-control form-search" type="text" placeholder="szukaj..."
+        span class="input-group-btn"
+          button class="btn btn-primary search-button" type="button"
+            i class="icon-search"
+      a href="#" zaloguj się
```
Comment (buffot) [inline source/_header.html.slim]: |  Zaloguj się

File: source/_header.html.slim [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+        p class="phone-number"
+          | tel. 223 009 400
+    div class="register-nav"
+      div class="input-group search-group"
+        input class="form-control form-search" type="text" placeholder="szukaj..."
+        span class="input-group-btn"
+          button class="btn btn-primary search-button" type="button"
+            i class="icon-search"
+      a href="#" zaloguj się 
+      a class="register" href='#' zarejestruj się
```
Comment (buffot) [inline source/_header.html.slim]: | Zarejstruj się

File: source/layouts/layout.html.slim [hunk 27]
```diff
@@ -2,11 +2,13 @@ doctype html
...
   head
     meta charset="utf-8" 
-
+    meta name="viewport" content="width=device-width, initial-scale=1.0"
     meta content="IE=edge,chrome=1" http-equiv="X-UA-Compatible" /
   
     title "The Middleman" 
-    == stylesheet_link_tag "normalize", "all" 
+    == stylesheet_link_tag 'bootstrap', 'all_styles'
+
```
Comment (buffot) [inline source/layouts/layout.html.slim]: remove empty line

File: source/stylesheets/all_styles.scss [hunk 28]
```diff
@@ -0,0 +1,5 @@
+@import "mediaqueries/media-header";
+@import "variables";
+@import "layout/_header";
+@import "components/_button";
+
```
Comment (buffot) [inline source/stylesheets/all_styles.scss]: remove empty lie

File: source/stylesheets/icons/fontello-codes.css [hunk 29]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (buffot) [inline source/stylesheets/icons/fontello-codes.css]: remove empty line

File: source/stylesheets/icons/fontello-ie7-codes.css [hunk 30]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (buffot) [inline source/stylesheets/icons/fontello-ie7-codes.css]: remove empty line

File: source/stylesheets/layout/_header.scss [hunk 31]
```diff
@@ -0,0 +1,162 @@
...
+//services navigation styling
+.services-nav {
+  display: flex;
+  justify-content: space-between;
+  margin: $base-margin 0;
+  p {
+    color: $grey;
+    font-weight: 400;
+    margin: $base-margin;
+  }
```
Comment (buffot) [inline source/stylesheets/layout/_header.scss:23]: Install and configure https://github.com/brigade/scss-lint for style check. Check all files

File: source/stylesheets/layout/_header.scss [hunk 32]
```diff
@@ -0,0 +1,162 @@
...
+      &:hover, &:focus {
+        text-decoration: none;
+      }
+      &:after {
+        content: "/";
+        font-weight: 900;
+        margin: 0 $base-margin;
+      }
+    }
+    .register {
```
Comment (buffot) [inline source/stylesheets/layout/_header.scss]: Improper naming `register` is more generic than `register-nav` so it shouldn't be nested. Change name `register` to be more specific

File: source/stylesheets/layout/_header.scss [hunk 33]
```diff
@@ -0,0 +1,162 @@
...
+      color: $turquoise;
+      text-decoration: none;
+    }  
+  }
+  .language {
+    width: 15px;
+    margin-bottom: $base-margin/2 
+  }
+}
+.register-nav a.no-after:after{
```
Comment (buffot) [inline source/stylesheets/layout/_header.scss]: This is really ugly it need to be improved. ``no-after` it makes no sense moreover it is nested

File: source/stylesheets/layout/_header.scss [hunk 34]
```diff
@@ -0,0 +1,162 @@
...
+    &:focus {
+    border-color: $turquoise;
+    outline-color: none;
+    }
+  }
+  .search-button {
+    @include search-button;
+  }
+}
+.text-slider-js div {
```
Comment (buffot) [inline source/stylesheets/layout/_header.scss:90]: this is only class for jQuery selectionc. It should not have any properties. This should be reduced only to `js-text-slider`  (`div` should be dropped) It should be prefixed on the beginning of the phrase

File: source/stylesheets/layout/_header.scss [hunk 35]
```diff
@@ -0,0 +1,162 @@
...
+  height: 100%;
+  margin-top: $base-margin*5;
+  }
+}
+
+// main navigation styling 
+.main-navigation {
+  display: flex;
+  justify-content: space-between;
+  margin-top: $base-margin*10;
```
Comment (buffot) [inline source/stylesheets/layout/_header.scss]: `10 *  $base-margin`

File: source/stylesheets/layout/_header.scss [hunk 36]
```diff
@@ -0,0 +1,162 @@
...
+      text-decoration: none;
+      padding-bottom: 4px;
+      color: $grey;
+    }
+    .location {
+      color: $turquoise;
+    }
+  }
+}
+// icons styling
```
Comment (buffot) [inline source/stylesheets/layout/_header.scss]: Should be moved to separate file `_icons.scss`

File: source/javascripts/all.js [hunk 37]
```diff
@@ -1,2 +1,11 @@
...
+//= require jquery
+
+$(document).ready(function(){
+  $(".text-slider-js div:gt(0)").hide();
+
+  setInterval(function() {
+    $(".text-slider-js div:first")
+    .fadeOut(1000).next().fadeIn(1000).end().appendTo(".text-slider-js");
+    }, 6000);
+});
```
Comment (buffot) [inline source/javascripts/all.js]: This is weird and too complicated. What is this for?

File: source/_header.html.slim [hunk 38]
```diff
@@ -0,0 +1,102 @@
...
+        img class="language" src='images/polish.png'  
+      a class="no-after" href="#" 
+        img class="language" src='images/english.png'
+  div class="row"
+    div class="nav-logo"
+      img class="logo" src="images/logo.png"
+  div class="row"
+    div class="container"
+      div class="primary-navigation"
+        ul class="main-navigation"
```
Comment (buffot) [inline source/_header.html.slim:38]: `primary-navigation` and `main-navigation` it is almost the same. Rename. Is `primary-navigation` div necessary?

File: source/_header.html.slim [hunk 39]
```diff
@@ -0,0 +1,102 @@
...
+          li class="nav-item"
+            a href='#' cennik 
+          li  class="nav-item"
+            a href='#' galeria
+          li class="nav-item"
+            a href='#' kontakt
+          li class="nav-item"
+            a class="drop-down-link location" href='#' lokalizacje
+            div class="drop-down-menu"
+              div class="menu-content"
```
Comment (buffot) [inline source/_header.html.slim]: Is this div necessary?  ul can have that class name (`menu-content`)

File: source/stylesheets/components/_button.scss [hunk 40]
```diff
@@ -0,0 +1,14 @@
+@mixin search-button {
```
Comment (buffot) [inline source/stylesheets/components/_button.scss]: This should be moved to _mixins.scss. Why this is mixin? It doesn't have any arguments or options to set.  It should be just simple class then

**Pull request merged**